source/dub/packagesuppliers/registry.d

```diff
@@ -86,8 +86,10 @@ class RegistryPackageSupplier : PackageSupplier {
 			m_metadataCache.remove(packageId);
 		}
 
-		auto url = m_registryUrl ~ NativePath("api/packages/infos?packages=[\"" ~
-				packageId ~ "\"]&include_dependencies=true&minimize=true");
+		auto url = m_registryUrl ~ NativePath("api/packages/infos");
+
+		url.queryString = "packages=[\"" ~
+				packageId ~ "\"]&include_dependencies=true&minimize=true";
```

Comment:
`packageId` doesn't really look like it would be escaped from the caller (like `fetchPackageRecipe(packageId, ...)` -> `getMetadata(packageId)`) so this could suffer from issues like having `&` in the packageId or manipulating the other arguments. Can you uri component escape the packageId?

Reply:
if this is the case, would this function solve the issue? https://dlang.org/phobos/std_uri.html#.encodeComponent

Reply:
yes that would be a good solution

Reply:
done